src/dialogflow/sessions_client_streaming.rs

```diff
@@ -147,10 +147,8 @@ impl SessionsClient {
                         streaming_recognize_result?.into_inner();
 
                     trace!("streaming_detect_intent_async_stream: entering loop");
-                    loop {
-                        if let Some(streaming_detect_intent_response) = response_stream.message().await? {
-                            yield streaming_detect_intent_response;
-                        }
+                    while let Some(streaming_detect_intent_response) = response_stream.message().await? {
```

Comment:
ok, I must admit I haven't worked with Rust for some time now. My Rust got little bit rusty :)

I am struggling little bit to understand why my original loop is blocking. My understanding is that once the code reaches this line:

`if let Some(streaming_detect_intent_response) = response_stream.message().await? {`

**await** will cause the future to yield the executor thread and the code will continue only once the future is ready to be polled again, i.e. when response message arrives. In this regard it seems pretty much as your code (though your code is shorter and nicer). I know you are probably right but I am missing the point here (probably too tired right now:) ). Can you provide additional explanation why my loops are blocking whereus yours are not?

Reply:
I thought it would work in theory as well for the same reasons you stated. Initially I explored other causes of our server receiving to accept new connections, but it was only with this change that it began to behave normally and handle other connections even when it was still waiting for messages in this stream.

That said though, it was with bidirectional speech-to-text that we encountered the issue.

Reply:
ok, examples seem to be working fine. I will accept this PR. Nevertheless it would be fine if you could come with some hard facts while `loop` is blocking whereus `while let Some(xxx)` loop is not. Once new version will be released it would be good to provide some relevant explanation what and how was fixed in release notes. So I think it would be handy to know more (I googled right now little bit but did not find any relevant answer).